annubes/tasks/task.py

```diff
@@ -250,32 +247,32 @@ def _build_trials_inputs(self) -> NDArray[np.float32]:
             for idx, m in enumerate(self.modalities):
                 if (self._modality_seq[n] != "catch") and (m in self._modality_seq[n]):
                     sel_value_in[n, idx] = self._rng.choice(self.stim_intensities[1:], 1)
-                sel_value_in[n, idx] = self._rescale(sel_value_in[n, idx], self.settings.rescaling_coeff)
+                sel_value_in[n, idx] = self._rescale(sel_value_in[n, idx], self.rescaling_coeff)
                 x[n, self._phases["input"], idx] = sel_value_in[n, idx]
                 x[n, self._phases["fix_time"], idx] = self._rescale(
-                    self.settings.fix_intensity,
-                    self.settings.rescaling_coeff,
+                    self.fix_intensity,
+                    self.rescaling_coeff,
                 )
             x[n, self._phases["input"], self.n_inputs - 1] = 1  # start cue
 
         # generate noise
-        alpha = self.settings.dt / self.settings.tau
-        noise_factor = self.settings.noise_std * np.sqrt(2 * alpha) / alpha
+        alpha = self.dt / self.tau
+        noise_factor = self.noise_std * np.sqrt(2 * alpha) / alpha
         noise = noise_factor * self._rng.normal(loc=0, scale=1, size=x.shape)
 
-        return x + self.settings.input_baseline + noise
+        return x + self.input_baseline + noise
 
     def _build_trials_outputs(self) -> NDArray[np.float32]:
         """Generate trial outputs."""
-        y = np.zeros((self._ntrials, len(self.time), self.settings.n_outputs), dtype=np.float32)
+        y = np.zeros((self._ntrials, len(self.time), self.n_outputs), dtype=np.float32)
         for i in range(self._ntrials):
-            if self.settings.iti is not None:
-                y[i, self._phases["iti"], :] = min(self.settings.output_behavior)
-            if self.settings.fix_time is not None:
-                y[i, self._phases["fix_time"], :] = min(self.settings.output_behavior)
+            if self.iti > 0:
+                y[i, self._phases["iti"], :] = min(self.output_behavior)
+            if self.fix_time > 0:
+                y[i, self._phases["fix_time"], :] = min(self.output_behavior)
```

Comment:
I think what I'm about to say is an issue for a separate PR and should be checked when creating the tests for this module in #17, but will put it down now already:

Are the `if`s even needed? I'm not 100% sure how to read it, but it seems to me that if they are 0, then no inputs will be set to the value given. If so, removing the `if` would make the code more readable.
Also, not sure whether in 144 and 145, where `self._phases["iti"]` and `self._phases["fix_time"]` are defined, it should be `<` instead of `<=`. (But maybe you are sure of this and it is correct now)
I think, in the tests, make sure there it is being tested whether input of 0 gives the expected result for these.

Reply:
Indeed let's check this separately when we develop the tests